src/esp/bindings/bindings.cpp

```diff
@@ -46,6 +50,13 @@ SceneNode* nodeGetter(T& self) {
     throw py::value_error{"feature not valid"};
   return &self.node();
 };
+
+struct PySensorSpec : SensorSpec {
+  py::dict noise_model_kwargs;
```

Comment:
Since this field is never used from C++, what about adding this attribute from Python side only? Pybind-exposed types don't have `__dict__` by default (just `__slots__`), defining `SensorSpec` with [py::dynamic_attributes()](https://pybind11.readthedocs.io/en/stable/classes.html#dynamic-attributes) would make that work. Then, in Python, you could

```py
sensor.noise_model_kwargs = {
    "gpu_device_id": self._sim.gpu_device
}

...

if hasattr(sensor, 'noise_model_kwargs'):
    ...
```

Alternatively, to avoid the `PySensorSpec`, using `def_property()` to define the `noise_model_kwargs` as a property that gets pulled out with `py::getattr()` if it's there, and returning an empty `py::dict` if not.

Reply:
Hm, and why not just this, once `SensorSpect` has the `__dict__`:

```py
sensor.gpu_device_id = self._sim.gpu_device

...
```

I think these could be added directly as instance attributes without losing any clarity.

Reply:
The `gpu_device_id` was really only meant as a argument to the noise model.  I will rewrite to clarify that.

Reply:
> Pybind-exposed types don't have __dict__ by default (just __slots__), defining SensorSpec with py::dynamic_attributes() would make that work. 

I had originally used this, but that felt wrong as its highly undiscoverable and seems harder to document properly.

Reply:
> Alternatively, to avoid the PySensorSpec, using def_property() to define the noise_model_kwargs as a property that gets pulled out with py::getattr() if it's there, and returning an empty py::dict if not.

I will give this a try